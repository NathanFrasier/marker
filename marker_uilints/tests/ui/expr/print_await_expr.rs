async fn foo() -> u8 {
    16
}

async fn bar() {
    let _print_await = foo().await;

    let future = foo();
    let _print_await = future.await;
}

fn main() {}
