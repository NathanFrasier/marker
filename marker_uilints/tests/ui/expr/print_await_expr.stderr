warning: print test
 --> $DIR/print_await_expr.rs:6:5
  |
6 |     let _print_await = foo().await;
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: Await(
              AwaitExpr {
                  data: CommonExprData {
                      _lifetime: PhantomData<&()>,
                      id: ExprId(..),
                      span: SpanId(..),
                  },
                  expr: Call(
                      CallExpr {
                          data: CommonExprData {
                              _lifetime: PhantomData<&()>,
                              id: ExprId(..),
                              span: SpanId(..),
                          },
                          operand: Path(
                              PathExpr {
                                  data: CommonExprData {
                                      _lifetime: PhantomData<&()>,
                                      id: ExprId(..),
                                      span: SpanId(..),
                                  },
                                  path: AstQPath {
                                      self_ty: None,
                                      path_ty: None,
                                      path: AstPath {
                                          segments: [
                                              AstPathSegment {
                                                  ident: Ident {
                                                      name: "foo",
                                                      span: Span {
                                                          source: File(
                                                              "$DIR/print_await_expr.rs",
                                                          ),
                                                          start: 73,
                                                          end: 76,
                                                      },
                                                  },
                                                  generics: SynGenericArgs {
                                                      args: [],
                                                  },
                                              },
                                          ],
                                      },
                                      target: Item(
                                          ItemId(..),
                                      ),
                                  },
                              },
                          ),
                          args: [],
                      },
                  ),
              },
          )
  = note: `#[warn(marker::test_lint)]` on by default

warning: print test
 --> $DIR/print_await_expr.rs:9:5
  |
9 |     let _print_await = future.await;
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: Await(
              AwaitExpr {
                  data: CommonExprData {
                      _lifetime: PhantomData<&()>,
                      id: ExprId(..),
                      span: SpanId(..),
                  },
                  expr: Path(
                      PathExpr {
                          data: CommonExprData {
                              _lifetime: PhantomData<&()>,
                              id: ExprId(..),
                              span: SpanId(..),
                          },
                          path: AstQPath {
                              self_ty: None,
                              path_ty: None,
                              path: AstPath {
                                  segments: [
                                      AstPathSegment {
                                          ident: Ident {
                                              name: "future",
                                              span: Span {
                                                  source: File(
                                                      "$DIR/print_await_expr.rs",
                                                  ),
                                                  start: 138,
                                                  end: 144,
                                              },
                                          },
                                          generics: SynGenericArgs {
                                              args: [],
                                          },
                                      },
                                  ],
                              },
                              target: Var(
                                  VarId(..),
                              ),
                          },
                      },
                  ),
              },
          )

warning: print test
  --> $DIR/print_await_expr.rs:11:5
   |
11 |     let _print_await = async { 18 }.await;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: Await(
               AwaitExpr {
                   data: CommonExprData {
                       _lifetime: PhantomData<&()>,
                       id: ExprId(..),
                       span: SpanId(..),
                   },
                   expr: Block(
                       BlockExpr {
                           data: CommonExprData {
                               _lifetime: PhantomData<&()>,
                               id: ExprId(..),
                               span: SpanId(..),
                           },
                           stmts: [],
                           expr: Some(
                               IntLit(
                                   IntLitExpr {
                                       data: CommonExprData {
                                           _lifetime: PhantomData<&()>,
                                           id: ExprId(..),
                                           span: SpanId(..),
                                       },
                                       value: 18,
                                       suffix: None,
                                   },
                               ),
                           ),
                           label: None,
                           is_unsafe: false,
                       },
                   ),
               },
           )

warning: 3 warnings emitted

